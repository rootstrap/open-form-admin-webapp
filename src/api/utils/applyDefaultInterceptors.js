const RS_API_KEY = 'RS-API-KEY';

export default function applyDefaultInterceptors(apiService) {
  apiService.requestInterceptors.use(request => {
    request.headers = { ...request.headers, [RS_API_KEY]: process.env.RS_API_KEY };
    return request;
  });
}
