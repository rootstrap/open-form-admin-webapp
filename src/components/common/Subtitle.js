import styled from 'styled-components';

export default styled.h2`
  font-weight: bold;
  letter-spacing: -0.3px;
  color: ${({ theme }) => theme.color.gray};
`;
