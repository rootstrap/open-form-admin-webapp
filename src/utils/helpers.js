import queryString from 'query-string';

export const parseInputErrors = error => {
  if (!error) {
    return;
  }
  if (Array.isArray(error)) {
    return error[0];
  }
  return error;
};

export const applyQueryParams = (url, params = {}) => {
  if (!Object.keys(params).length) {
    return url;
  }
  const queryParams = queryString.stringify(params);
  return `${url}?${queryParams}`;
};
