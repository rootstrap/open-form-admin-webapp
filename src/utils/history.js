import createBrowserHistory from 'history/createBrowserHistory';
import createMemoryHistory from 'history/createMemoryHistory';

let instance;

class History {
  constructor() {
    if (!instance) {
      instance = process.env.BROWSER ? createBrowserHistory() : createMemoryHistory();
    }
    return instance;
  }
}

export default new History();
