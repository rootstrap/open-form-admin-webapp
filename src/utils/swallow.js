const swallow = (fail, fn) => (...args) => {
  try {
    return fn(...args);
  } catch (error) {
    return fail(error, ...args);
  }
};

export default swallow;
