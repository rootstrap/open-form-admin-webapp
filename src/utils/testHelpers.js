import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import theme from 'constants/theme';

export function renderWithTheme(component) {
  return TestRenderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}
