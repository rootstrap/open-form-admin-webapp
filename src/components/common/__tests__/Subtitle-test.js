import React from 'react';

import { renderWithTheme } from 'utils/testHelpers';
import Subtitle from '../Subtitle';

it('renders correctly', () => {
  const testRenderer = renderWithTheme(<Subtitle>info text</Subtitle>);
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
