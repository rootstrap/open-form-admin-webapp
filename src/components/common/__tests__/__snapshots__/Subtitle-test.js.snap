// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  font-weight: bold;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  color: #4f4f4f;
  margin-bottom: 0;
}

<h2
  className="c0"
>
  info text
</h2>
`;
